### PR TITLE
feat(content-conformance): implement DataFile

### PR DESCRIPTION
**Pull request by BRKalow**
🎟️ [Asana Task](https://app.asana.com/0/1100423001970639/1203332374839498/f)

---

## Description

Implement the `DataFile` class. Supports `.json` and `.yaml` files. The resulting object is immutable and accessible via `.content()` for use in rules.

## PR Checklist 🚀

- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Write a useful description (above) to give reviewers appropriate context.
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: packages/content-conformance/src/engine.ts [hunk 1]
```diff
@@ -39,6 +39,33 @@ export class ContentConformanceEngine {
...
+      const contents = await fs.promises.readFile(fullPath, 'utf-8')
+
+      const file = new DataFile({
+        cwd: this.opts.root,
+        path: String(filepath),
+        value: contents,
+      })
+      this.dataFiles.push(file)
+    }
+  }
```
Comment (BRKalow) [inline packages/content-conformance/src/engine.ts:67]: `loadDataFiles` and `loadContentFiles` are almost identical. I think we can likely consolidate the two methods going forward 💭

File: packages/content-conformance/src/data-file.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      default: {
+        console.warn(
+          `[content-conformance] unable to parse data file with extension: ${this.extname}`
+        )
+      }
+    }
+  }
+}
+
+export function isContentFile(file: unknown): file is DataFile {
```
Comment (BRKalow) [inline packages/content-conformance/src/data-file.ts:69]:
```suggestion
export function isDataFile(file: unknown): file is DataFile {
```

File: packages/content-conformance/src/data-file.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    get(target, property) {
+      return typeof target[property] === 'object'
+        ? immutable(target[property])
+        : target[property]
+    },
+    set() {
+      throw new Error('Unable to modify object')
+    },
+  })
+}
```
Comment (BRKalow) [inline packages/content-conformance/src/data-file.ts:21]: Use a [`Proxy`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) to make an object "immutable". It'll throw if someone attempts to set a property via the defined `set()` handler.

File: packages/content-conformance/src/__tests__/data-file.test.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  - title: About
+    path: /about
+  `,
+      path: '/nav.yaml',
+    })
+
+    expect(file.contents()).toEqual(fixture)
+  })
+
+  test('parsed object is immutable', () => {
```
Comment (thiskevinwang) [inline packages/content-conformance/src/__tests__/data-file.test.ts:36]: 🙅 ✨

**Review by thiskevinwang — APPROVED** [review]
✅ Very straightfoward!

**Comment by BRKalow** [comment]
@nayres I'm gonna merge this but feel free to review / take a look! I can address any comments in a follow-up.

**Pull request merged**